### ISSUE TITLE
Browser freezes for a few seconds when dropping file

### ISSUE THREAD

**Issue by ddoriazyp** [Wildhoney/ngDroplet#46]
I'm working on a web app that allows users to upload files up to 250MB. The sample file I'm working with is around 88MBs and whenever I drag and drop, the browser noticeably freezes for a few seconds before it's responsive again. I'm using Chrome Version 48.0.2564.97 (64-bit).

**Comment by Wildhoney** [comment]
Are you using the `droplet-preview` directive? As that's the only point in which [we read the file data](https://github.com/Wildhoney/ngDroplet/blob/master/components/ngDroplet.js#L1035) &mdash; but it shouldn't attempt to read it unless it's an image.

**Comment by ddoriazyp** [comment]
Yes, I'm using the `droplet-preview` directive. Btw, it's a video file I'm uploading and am not sure if that would be an issue. Aside from the temporary freezing issue, everything else works fine and am able to upload whatever files to my backend.

**Comment by Wildhoney** [comment]
Hmm... that shouldn't matter.

Can you replicate it [on the demo site](http://ng-droplet.herokuapp.com/)?

I've tried with a 512MB JPEG that I have of Mars, and I don't notice any slowdown when dropping it onto there.

**Comment by ddoriazyp** [comment]
My app accepts video files but the demo site apparently only accepts images so I took a ~90MB file, changed the extension to .jpg and dropped the file and I was able to duplicate what I'm experiencing. After I drop it, I clicked on the 'Add Files...' button and it doesn't respond until a few seconds later.

**Comment by Wildhoney** [comment]
I'm unable to replicate that.

Could you please try a different browser on the demo site, and see if it's something you can consistently reproduce across a handful of browsers?

**Comment by ddoriazyp** [comment]
I tested the same file on Safari and Firefox Developer edition and got similar results, more pronounced on Firefox. However, I tried it on larger media files and didn't get the issue. I'm starting to think it's an issue with certain files that may be causing this. 

Having confirmation that you aren't running into these issues, I'm confident going forward that this should be an issue easily handled on my side. Thanks for the quick responses and for an awesome directive.
